### PR TITLE
Setting failIfNoResultsValue to true for all types of jobs.

### PR DESCRIPTION
**Pull request by amir-waheed**
Fixes issue #3269.

Signed-off-by: Abeer Waheed <amir2@ualberta.ca>

### CODE REVIEW

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -678,9 +678,6 @@ def post(output_name) {
 			}
 
 			boolean failIfNoResultsValue = true
-			if (params.PARALLEL && params.PARALLEL != "None") {
-				failIfNoResultsValue = false
-			}
 			step([$class: "TapPublisher", testResults: "**/*.tap", outputTapToConsole: false, failIfNoResults: failIfNoResultsValue])
```
Comment (smlambert) [inline buildenv/jenkins/JenkinsfileBase:681]: Since its only used in line 681, likely do not need to declare a boolean in line 680, just set failIfNoResults: true in line 681.

**Review by smlambert — COMMENTED** [review]
Added a couple of suggested changes, and once pushed, this PR will need a few Grinder jobs to test behaviour (testing PARALLEL=None and PARALLEL=Dynamic and PARALLEL=Subdir or NodesByIterations).

**Comment by amir-waheed** [comment]
> Added a couple of suggested changes, and once pushed, this PR will need a few Grinder jobs to test behaviour (testing PARALLEL=None and PARALLEL=Dynamic and PARALLEL=Subdir or NodesByIterations).

Thank you for the suggestions. I have committed the suggested changes. I will start with the tests now.

**Comment by amir-waheed** [comment]
@smlambert following is the result of running one of the tests, it seems it that timeout exceeded: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3396/console

**Comment by smlambert** [comment]
Oh yes, for Grinder 3396, there are no zos machines attached to this Jenkins server, so PLATFORM=zos_390-64 will always exit/fail with "Could not find any nodes with 'ci.role.test&&hw.arch.s390x&&sw.os.zos' label" message.

**Comment by amir-waheed** [comment]
Thank you @smlambert. Following are the link for the tests that I ran:

**Parallel = None** 
Link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3399/
Status: Successful 

**Parallel = Dynamic**
Link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3400/
Status: Successful 

**Parallel = Subdir** 
Link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3401/console
Status: Failed (Unstable build)
Link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3404/console
Status: Failed (Unstable build)

**Parallel = NodesByIteration** 
Link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3402/
Status: Successful

**Comment by smlambert** [comment]
Adding more Grinder examples, for some additional testing:

Running PARALLEL=Subdir with BUILD_LIST=system and TARGET=sanity.system 
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3406

**Comment by llxia** [comment]
Please also update this line https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase#L988. Thanks

related: https://github.com/adoptium/aqa-tests/issues/3077

**Comment by llxia** [comment]
Parallel = Dynamic: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3504/

**Review by llxia — APPROVED** [review]
LGTM
